liblouis/utils.c

```diff
@@ -136,25 +136,28 @@ _lou_charHash(widechar c) {
 	return (unsigned long int)c % HASHNUM;
 }
 
-char *EXPORT_CALL
-_lou_showString(widechar const *chars, int length) {
+const char *EXPORT_CALL
+_lou_showString(widechar const *chars, int length, int forceHex) {
 	/* Translate a string of characters to the encoding used in character
 	 * operands */
-	int charPos;
-	int bufPos = 0;
 	static char scratchBuf[MAXSTRING];
+	int bufPos = 0;
 	scratchBuf[bufPos++] = '\'';
-	for (charPos = 0; charPos < length && bufPos < (MAXSTRING - 2); charPos++) {
-		if (chars[charPos] >= 32 && chars[charPos] < 127)
-			scratchBuf[bufPos++] = (char)chars[charPos];
-		else {
+
+	for (int charPos = 0; (charPos < length) && (bufPos < (MAXSTRING - 2));
+			charPos += 1) {
+		widechar c = chars[charPos];
+
+		if (!forceHex && (c >= 0X20) && (c < 0X7F)) {
```

Comment:
Don't we have constants for these magic numbers?

Reply:
For me these don't need to be constants. Would maybe even make it less readable. But maybe a comment above the rule that says this means "c inside the ASCII range" would be useful.

Reply:
OK, fine